go/tao/kvm_coreos_factory.go

```diff
@@ -332,52 +386,61 @@ func getRandomFileName(n int) string {
 	return hex.EncodeToString(nameBytes)
 }
 
+// Spec returns the specification used to start the hosted vm.
+func (kcc *KvmCoreOSContainer) Spec() HostedProgramSpec {
+	return kcc.spec
+}
+
 var nameLen = 10
 
-// Launch launches a QEMU/KVM CoreOS instance, connects to it with SSH to start
+// Start launches a QEMU/KVM CoreOS instance, connects to it with SSH to start
 // the LinuxHost on it, and returns the socket connection to that host.
-func (lkcf *LinuxKVMCoreOSFactory) Launch(imagePath string, args []string, uid, gid int) (io.ReadWriteCloser, HostedProgram, error) {
+func (kcc *KvmCoreOSContainer) Start() (channel io.ReadWriteCloser, err error) {
+
 	// The args must contain the directory to write the linux_host into, as
 	// well as the port to use for SSH.
-	if len(args) != 3 {
-		glog.Errorf("Expected %d args, but got %d", 3, len(args))
-		for i, a := range args {
+	if len(kcc.spec.Args) != 3 {
+		glog.Errorf("Expected %d args, but got %d", 3, len(kcc.spec.Args))
+		for i, a := range kcc.spec.Args {
 			glog.Errorf("Arg %d: %s", i, a)
 		}
-		return nil, nil, errors.New("KVM/CoreOS guest Tao requires args: <linux_host image> <temp directory for linux_host> <SSH port>")
+		err = errors.New("KVM/CoreOS guest Tao requires args: <linux_host image> <temp directory for linux_host> <SSH port>")
+		return
 	}
 	// Build the new Config and start it. Make sure it has a random name so
 	// it doesn't conflict with other virtual machines. Note that we need to
 	// assign fresh local SSH ports for each new virtual machine, hence the
 	// mutex and increment operation.
+	// TODO(kwalsh) what mutex and increment?
```

Comment:
Looks like this is a stale part of the comment; feel free to delete. The idea was that there might be multiple simultaneous requests for VM creation, and the ports needed to be unique for each, so there needed to be an atomic increment operation.